build.sbt

```diff
@@ -64,18 +64,34 @@ lazy val core = projectMatrix
   .settings(
     moduleName := "scalafix-core",
     buildInfoSettingsForCore,
-    libraryDependencies ++= List(
-      scalameta,
-      googleDiff,
-      collectionCompat
+    scalacOptions --= (
+      if (isScala3.value)
+        Seq("-P:semanticdb:synthetics:on")
+      else Nil
```

Comment:
this is repeated in many places, can we move it to a settings block covering all current usages?

Reply:
factored out